SqlServer/Import-BacpacToSqlServer/Import-BacpacToSqlServer.psm1

```diff
@@ -160,35 +212,53 @@ function Import-BacpacToSqlServer
 
 
 
-        # And now we get to actually importing the database after constructing a connection string with validated details.
-        $ConnectionString = "Data Source=$DataSource;Initial Catalog=$DatabaseName;Integrated Security=True;"
-        & "$sqlPackageExecutablePath" /Action:Import /SourceFile:"$BacpacPath" /TargetConnectionString:"$ConnectionString"
+        $parameters = @(
+            '/Action:Import',
+            "/TargetServerName:`"$DataSource`"",
+            "/TargetDatabaseName:`"$DatabaseName`"",
+            '/TargetTrustServerCertificate:True',
+            '/TargetTimeout:0',
+            "/SourceFile:`"$BacpacPath`""
+        )
+
+        if (![string]::IsNullOrEmpty($Username) -and ![string]::IsNullOrEmpty($Password))
+        {
+            $parameters.Add("/TargetUser:`"$Username`"")
+            $parameters.Add("/TargetPassword:`"$Password`"")
+        }
+
+        # And now we get to actually importing the database after setting up all the necessary parameters.
+        & "$sqlPackageExecutablePath" @parameters
 
 
 
         # Importing the database was successful.
         if ($LASTEXITCODE -eq 0)
         {
-            # If there was a database with an identical name on the target server, we'll swap it with the one we've just imported.
+            # If there was a database with an identical name on the target server, we'll swap it with the one we've just
+            # imported.
             if ($databaseExists)
             {
                 $server = New-Object ('Microsoft.SqlServer.Management.Smo.Server') $DataSource
                 $server.KillAllProcesses($originalDatabaseName)
```

Comment:
After a successful import, it crashed out here with the following exception:
> Import-BacpacToSqlServer : Exception calling "KillAllProcesses" with "1" argument(s): "Failed to connect to server .."

I guess you have to pass the username/password to the `$server` constructor too?